Modules/Microsoft365DSC/Modules/M365DSCUtil.psm1

```diff
@@ -1054,7 +1054,7 @@ function Export-M365DSCConfiguration
         foreach ($dependency in $outdatedOrMissingAssemblies)
         {
             Write-Host "Updating dependency {$($dependency.ModuleName)} to version {$($dependency.RequiredVersion)}..." -NoNewline
-            Install-Module $dependency.ModuleName -RequiredVersion $dependency.RequiredVersion -Force | Out-Null
+            Install-Module $dependency.ModuleName -RequiredVersion $dependency.RequiredVersion -Force -Scope 'AllUsers'  | Out-Null
```

Comment:
This will require that the user is running as a local admin, else it will throw an error. Maybe it is a good idea to check if the user has these permissions and if not, show a nice message:

You can use this code to check if the user has admin permissions:
`([Security.Principal.WindowsPrincipal] [Security.Principal.WindowsIdentity]::GetCurrent()).IsInRole([Security.Principal.WindowsBuiltInRole]::Administrator)`

Reply:
Great suggestion. Code updated to add this check and throw a meaningful error if the user is not an admin. Thanks

Reply:
Just a further suggestion from a code refactoring context: Encapsulate this one in a function `Test-AdminAccountUsage` to add additional context information e.g. "You need to run this code as Admin"

```powershell

function Test-AdminAccountUsage{
 $adminrole = ([Security.Principal.WindowsPrincipal] [Security.Principal.WindowsIdentity]::GetCurrent()).IsInRole([Security.Principal.WindowsBuiltInRole]::Administrator)
 if($adminrole){
  Write-Verbose "This command needs Admin Permissions"
 }
 return $adminrole
}
```

Reply:
While I normally like to wrap as much stuff in functions to promote re-use and reduce duplication of code, this admin check will almost always need to be wrapped inside an if statement itself to prevent further execution of code when found to be false. Because this function would end up being a simply if check itself, it doesn't make sense in my opinion to have if further wrapped in conditional logic.

E.g., 

if (Test-AdminAccountUsage)
{
    #do Work
}
[...]

function Test-AdminAccountUsage{
 $adminrole = ([Security.Principal.WindowsPrincipal] [Security.Principal.WindowsIdentity]::GetCurrent()).IsInRole([Security.Principal.WindowsBuiltInRole]::Administrator)
 if($adminrole){
  Write-Verbose "This command needs Admin Permissions"
 }
 return $adminrole
}

Reply:
That is a valid point. Having a second handling within the function is nonsense. Only benefit would be a better readability...

Could we try to self elevate the process to gain a bit more usability?

```
function Test-AdminAccountUsage{
return ([Security.Principal.WindowsPrincipal] [Security.Principal.WindowsIdentity]::GetCurrent()).IsInRole([Security.Principal.WindowsBuiltInRole]::Administrator)
}
```